src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -592,96 +592,102 @@ namespace smt::noodler {
-        //      - if a1 or a2 is only empty word, then only |L| != |R| can hold
-        //      - we check a1!char != a2!char instead of a1 != a2 and then we add at the end that a1!char/a2!char
-        //        should be equal to one of the symbols of a1/a2
-        //      - we also work with maximally substituted a1/a2, so that if we have multiple disequations that
-        //        somehow depend on each other, the dependency stays
-
-        // we flatten the subtitution map, so that all the vars a1/a2 whose disequality we will check
-        // are substituted by maximal application of substitution_map
+        // the following functions (getting formula for transformations) assume that we have flattened substitution map
         solution.flatten_substition_map();
 
-        // here we save all a1/a2 whose disequation is checked, so that we can create the formula that 
-        // each of these vars should be one of the symbols from its automaton
-        std::set<BasicTerm> a_vars;
-        // get the "!char" version of a variable
-        auto get_char_var = [](const BasicTerm& var) -> BasicTerm { return BasicTerm(BasicTermType::Variable, var.get_name().encode() + "!char"); };
-
-        // conjuncts of the final length formula
-        std::vector<LenNode> conjuncts;
+        // add formula for transformations
+        conjuncts.push_back(transformation_formula());
 
-        for(const auto& pr: dis_len) {
-            // formula for |x1| == |x2|
-            LenNode x1x2_same_lengths = pr.second.first;
-            // formula for |L| != |R|
-            LenNode LR_diff_lengths = pr.second.second;
+        return LenNode(LenFormulaType::AND, conjuncts);
+    }
 
-            // variables a1 and a2 where we check a1 != a2
-            BasicTerm a1 = pr.first.first;
-            BasicTerm a2 = pr.first.second;
+    LenNode DecisionProcedure::transformation_formula() {
+        std::vector<LenNode> result_conjuncts;
+        for (const std::tuple<BasicTerm, BasicTerm, TransformationType>& transf : transformations) {
+            BasicTerm result = std::get<0>(transf);
+            BasicTerm argument = std::get<1>(transf);
+            TransformationType type = std::get<2>(transf);
+            auto to_code_var = [](const BasicTerm& var) -> BasicTerm { return BasicTerm(BasicTermType::Variable, var.get_name() + "!to_code"); };
+            auto process_one_symbol_words = [this, &to_code_var](const BasicTerm& string_var, const BasicTerm &int_var) -> LenNode {
+                std::vector<LenNode> to_code_disjunction;
+                for (Mata::Symbol s : Mata::Strings::get_one_symbol_words(*solution.aut_ass.at(string_var))) {
+                    if (s == OTHER_SYMBOL) {
+                        // if s == minterm => do something else
+                        std::vector<LenNode> conjuncts_here{LenNode(LenFormulaType::LEQ, {0, to_code_var(string_var)}), LenNode(LenFormulaType::LEQ, {to_code_var(string_var), 196607})};
```

Comment:
Replace the magic constant 196607 with a constexpr constant, similar to the `OTHER_SYMBOL` constant.

Reply:
Replaced with call to Z3 `max_char()`